### DISCUSSION TITLE
LTSP command failed: test MAC_ADDRESS= != MAC_ADDRESS= Aborting ltsp

### DISCUSSION THREAD

**Discussion by dcameron2** [ltsp/ltsp#225] [Q&A]
Hi, I hope I'm in the right place......

I am trying to get an LTSP server running.  Following the instructions for a chrootless installation I had a working server on my "prototype" machine in minutes. Yeah!

Unfortunately, following the same procedure on my (substantially similar) production machine I get the following error when executing 'ltsp dnsmasq' 

LTSP command failed: test MAC_ADDRESS= != MAC_ADDRESS=
Aborting ltsp

I am at a loss to understand what causes this or the difference between machines that could be the problem. No luck googling...

Any hints or help is appreciated!

I am running Ubuntu 18.04 server with lubuntu desktop environment

Thanks!

**Comment by alkisg** [comment]
Happy new year to all,

this is the line that fails:
https://github.com/ltsp/ltsp/blob/master/ltsp/common/ltsp/55-config.sh#L289

Does the server have an IP? Can you run the following commands at that point, and paste the output?

```shell
ip -o link show
ip -o a
echo /sys/class/net/*/device
```

**Comment by dcameron2** [comment]
Happy New Year, and thanks for the quick reply.  I have copied the 
output below, but meanwhile I have an hypothesis as to the cause of the 
problem.

I suspect the problem may be with the dhcp.  My understanding of network 
protocols is sketchy at best, but here is what I have observed:

In the past I have been running this computer with a static ip of 
192.168.2.2 (the router ip is 192.168.2.1).  When I installed 18.04 I 
had no need of a static ip anymore, so I left it as dhcp.  Meanwhile the 
router's ip pool starting and ending addresses are, respectively,  
192.168.2.20 and 192.168.2.240. Notwithstanding these facts, the 
computer's ip address is currently 192.168.2.2.

My understanding is that when the router received the dhcp request it 
recognized the mac address and hostname and assigned 192.168.2.2 to the 
computer as a static ip. The computer meanwhile is assuming this address 
was dynamically assigned.  I don't understand the details of what 
problems this might cause, but I am surprised that I have been able to 
run this way for over a year with no noticeable problems!

My router is an ASUS RT-AC68U, and it includes the ability to manually 
assign ip's around the dhcp list.  This computer's hostname and mac 
address are on that list.  I believe that this is a feature of the 
router that gives the ability to set static addressing while leaving the 
computer's settings as dhcp.  In any case, I removed the computer host 
name/mac address from this list, and the router immediately reassigned 
the computer ip to 192.168.2.38.  So far so good.

Unfortunately, after rebooting and re-running 'ltsp dnsmasq' I get the 
same error. Below are the outputs of the commands you requested.  The 
first is before the router change (computer ip 192.168.2.2) The second 
is after the change (computer ip 192.168.2.38)

Thanks again for the quick reply!

Dave C.

BEFORE:

david@oak:~$ ip -o link show
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode 
DEFAULT group default qlen 1000\    link/loopback 00:00:00:00:00:00 brd 
00:00:00:00:00:00
2: enp3s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel 
state UP mode DEFAULT group default qlen 1000\ link/ether 
90:e6:ba:e3:87:9e brd ff:ff:ff:ff:ff:ff
4: tun0: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc 
fq_codel state UNKNOWN mode DEFAULT group default qlen 100\    link/none
david@oak:~$ ip -o a
1: lo    inet 127.0.0.1/8 scope host lo\       valid_lft forever 
preferred_lft forever
1: lo    inet6 ::1/128 scope host \       valid_lft forever 
preferred_lft forever
2: enp3s0    inet 192.168.2.2/24 brd 192.168.2.255 scope global dynamic 
enp3s0\       valid_lft 49334sec preferred_lft 49334sec
2: enp3s0    inet6 fe80::92e6:baff:fee3:879e/64 scope link \ valid_lft 
forever preferred_lft forever
4: tun0    inet 10.16.10.6 peer 10.16.10.5/32 scope global tun0\       
valid_lft forever preferred_lft forever
4: tun0    inet6 fe80::ccf7:cc69:6e23:3962/64 scope link stable-privacy 
\       valid_lft forever preferred_lft forever
david@oak:~$ echo /sys/class/net/*/device
/sys/class/net/enp3s0/device

AFTER:

david@oak:~$ ip -o link show
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode 
DEFAULT group default qlen 1000\    link/loopback 00:00:00:00:00:00 brd 
00:00:00:00:00:00
2: enp3s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel 
state UP mode DEFAULT group default qlen 1000\ link/ether 
90:e6:ba:e3:87:9e brd ff:ff:ff:ff:ff:ff
3: tun0: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc 
fq_codel state UNKNOWN mode DEFAULT group default qlen 100\    link/none
david@oak:~$ ip -o a
1: lo    inet 127.0.0.1/8 scope host lo\       valid_lft forever 
preferred_lft forever
1: lo    inet6 ::1/128 scope host \       valid_lft forever 
preferred_lft forever
2: enp3s0    inet 192.168.2.38/24 brd 192.168.2.255 scope global dynamic 
enp3s0\       valid_lft 85687sec preferred_lft 85687sec
2: enp3s0    inet6 fe80::92e6:baff:fee3:879e/64 scope link \ valid_lft 
forever preferred_lft forever
3: tun0    inet 10.7.10.6 peer 10.7.10.5/32 scope global tun0\       
valid_lft forever preferred_lft forever
3: tun0    inet6 fe80::8312:9cb2:3327:2db6/64 scope link stable-privacy 
\       valid_lft forever preferred_lft forever
david@oak:~$ echo /sys/class/net/*/device
/sys/class/net/enp3s0/device

**Comment by alkisg** [comment]
I think the problem is either in tun or in ipv6; not in dynamic/static IP.

Can you try the following?

`sudo nano /usr/share/ltsp/common/ltsp/55-config.sh`
And add `set -x` in line 248:

```shell
network_vars() {
    local ip _dummy
    set -x
```

Then save and run `sudo ltsp dnsmasq` again. It will produce a lot of debug output, please upload all it here.
